sdk/core/azure_core_amqp/src/value.rs

```diff
@@ -103,6 +111,37 @@ impl AmqpDescribed {
             value: value.into(),
         }
     }
+
+    pub fn descriptor(&self) -> &AmqpDescriptor {
+        &self.descriptor
+    }
+
+    pub fn value(&self) -> &AmqpValue {
+        &self.value
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct AmqpComposite {
+    pub descriptor: AmqpDescriptor,
+    pub value: AmqpList,
+}
+
+impl AmqpComposite {
+    pub fn new(descriptor: impl Into<AmqpDescriptor>, value: impl Into<AmqpList>) -> Self {
```

Comment:
Is the `impl Into<T>` overkill here? That's mainly meant for common types like `String`.

Reply:
I've started using `impl Into<T>` pretty much everywhere because it means that I can do things like passing in a either a string or an integer to `AmqpComposite::new`, and I can also do things like passing in a `Vec<AmqpValue>` to the constructor without having to force the caller to type `.into()` all over the place.

Basically it moves the work of calling `.into()` to the implementation rather than forcing the consumer of the type to call `.into()`. And tbh, I hate typing `.into()` all over the place :).

If moving `.into()` calls into the implementation of a function is not considered idiomatic, then so be it.

Reply:
FWIW, one of the other really *nice* aspects of doing the `Into<EventData>` thingy is that you can do:

```rust
message_producer.send_event("This is a test message", None);
message_producer.send_event(vec![1,2,3,4], None);
```

which IMHO is a nicer developer experience from:
```rust
message_producer.send_event("This is a test message".into(), None);
message_producer.send_event(vec![1,2,3,4].into(), None);
```

that's because the `.into()` doesn't contribute to the developer experience, in fact it detracts from it because you try the thing you want to try (just adding the message body), then you get an error about a type mismatch, then you have to add the `.into()` to resolve it.

Reply:
No, that's fine to leave it as-is. I just wasn't expecting concrete types to often-enough have `Into<AmqpComposite>` impls, but if enough to make it worth it, great!